pkg/controller/endpointslicemirroring/endpointslicemirroring_controller.go

```diff
@@ -273,6 +278,36 @@ func (c *Controller) syncEndpoints(key string) error {
 		return err
 	}
 
+	endpoints, err := c.endpointsLister.Endpoints(namespace).Get(name)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return c.deleteMirroredSlices(namespace, name)
```

Comment:
just to double-check, deleteMirroredSlices returns success (nil error) if there are no mirrored endpointslices for the given namespace/name, right?

Reply:
Correct. There are 2 kinds of errors that can be returned here:

1. An error resulting from `endpointSliceLister.EndpointSlices(namespace).List(esLabelSelector)`
2. An error resulting from `r.client.DiscoveryV1beta1().EndpointSlices(namespace).Delete(context.TODO(), endpointSlice.Name, metav1.DeleteOptions{})`